src/views/ApartmentApp/components/StaffMenu/index.js

```diff
@@ -65,22 +65,17 @@ const StaffMenu = ({ userProfile, authentication }) => {
     let applicationsForCsv = [];
     let applicantsForCsv = [];
     let apartmentChoicesForCsv = [];
-    applicationDetailsArray.forEach((applicationDetails) => {
+    applicationDetailsArray.forEach(({ Applicants, ApartmentChoices, ...applicationDetails }) => {
       // Only add the applications that have been submitted
       if (applicationDetails.DateSubmitted) {
-        let { Applicants, ApartmentChoices, ...filteredApplicationDetails } = applicationDetails;
-        applicationsForCsv.push(filteredApplicationDetails);
-
-        Applicants.forEach((applicant) => {
-          if (applicant.ApplicationID) {
-            applicantsForCsv.push(applicant);
-          } else {
-            applicantsForCsv.push({
-              ApplicationID: applicationDetails.ApplicationID,
-              ...applicant,
-            });
-          }
-        });
+        applicationsForCsv.push(applicationDetails);
```

Comment:
Would it be better to do this via something like this instead of pushing to an array, followed by using that array in the `setApplicationJsonArray()`?

```
setApplicationJsonArray(applicationDetailsArray.filter((applicationDetails) => (
    applicationDetails?.DateSubmitted)
  )
  .map(({ Applicants, ApartmentChoices, ...applicationDetails }) => {
    // do the stuff with the applicants and apartmentchoices here
    return applicationDetails;
  }
)
```

Reply:
I have begun drafting this in a new branch, to test it out